### PR TITLE
[DCA-18117] Ocaml changes to expose c methods & Defragmentation logic

### PR DESCRIPTION
**Pull request by mgarewal-r7**
Added the following methods to the ocaml generator: defragment, free_bytes, used_bytes. 
Defragmenting logic removes unused bytes inside hives.
See test plan - https://wiki.corp.rapid7.com/display/PM/Test+Plan+for+DCA-18117+-+Defragmentation+for+Hivex+Testing

### CODE REVIEW

File: generator/generator.ml [hunk 1]
```diff
@@ -634,9 +650,6 @@ let check_functions () =
         if contains_uppercase n then
           failwithf "%s param/ret %s should not contain uppercase chars"
             name n;
-        if String.contains n '-' || String.contains n '_' then
-          failwithf "%s param/ret %s should not contain '-' or '_'"
-            name n;
```
Comment (skhare-r7) [inline generator/generator.ml]: Why remove this?

File: generator/generator.ml [hunk 2]
```diff
@@ -683,7 +696,7 @@ let check_functions () =
   (* Check short descriptions. *)
   List.iter (
     fun (name, _, shortdesc, _) ->
-      if shortdesc.[0] <> Char.lowercase shortdesc.[0] then
+      if shortdesc.[0] <> Char.lowercase_ascii shortdesc.[0] then
```
Comment (skhare-r7) [inline generator/generator.ml:702]: ?

File: lib/hivex-internal.h [hunk 3]
```diff
@@ -347,7 +348,7 @@ extern int _hivex_get_values (hive_h *h, hive_node_h node, hive_value_h **values
 
 /* These limits are in place to stop really stupid stuff and/or exploits. */
 #define HIVEX_MAX_SUBKEYS      200000
-#define HIVEX_MAX_VALUES       100000
+#define HIVEX_MAX_VALUES        55000
```
Comment (skhare-r7) [inline lib/hivex-internal.h]: Dont change this, keep the old value

File: lib/write.c [hunk 4]
```diff
@@ -342,7 +342,7 @@ mark_block_unused (hive_h *h, size_t offset)
     //Duplicated vks found in nk, ignore double deletion request
     return;
   }
-
+
```
Comment (skhare-r7) [inline lib/write.c]: Fix whitespace

File: lib/hivex-internal.h (hunk 3 above)
Comment (mgarewal-r7) [reply lib/hivex-internal.h]: Are you sure? The change was merged in from your defragment branch.

File: generator/generator.ml (hunk 2 above)
Comment (mgarewal-r7) [reply generator/generator.ml:702]: lowercase is no longer compatible with latest ocaml version, its now lowercase_ascii

File: generator/generator.ml (hunk 1 above)
Comment (mgarewal-r7) [reply generator/generator.ml]: Removed check as it prevented creating methods with argument names including a "_"

**Review by skhare-r7 — APPROVED** [review]

**Pull request merged**